pkg/services/internal/whatsonchain/service.go

```diff
@@ -358,3 +358,23 @@ func (woc *WhatsOnChain) IsUtxo(ctx context.Context, scriptHash string, outpoint
 
 	return status.IsUtxo, nil
 }
+
+func (woc *WhatsOnChain) GetStatusForTxIDs(ctx context.Context, txIDs []string) (*wdk.GetStatusForTxIDsResult, error) {
+	if len(txIDs) == 0 {
+		return nil, fmt.Errorf("no txIDs provided")
+	}
+
+	url, err := txsStatusURL(woc.url)
+	if err != nil {
+		return nil, fmt.Errorf("failed to build URL: %w", err)
+	}
+
+	results, err := woc.getStatusForTxIDs(ctx, url, txIDs)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get status for txIDs: %w", err)
+	}
+	if results == nil {
```

Comment:
```suggestion
	if len(results) == 0 {
```

If getStatusForTxIDs returns a pointer but with an empty Results array (but not nil), this would fall through and appear as success, even if no data was returned. Meanwhile, other implementations (see Bitails) explicitly return an error if no results are found. This may cause inconsistent handling.

Reply:
invalid argument: results (variable of type *wdk.GetStatusForTxIDsResult) for built-in len

Reply:
I'm wondering should we return somehow information that it is "not found" situation. 🤔 
For example like this:
```golang
return nil, fmt.Errorf("no status found for provided txIDs: %w", wdk.NotFoundError)
```

Reply:
ok i added check for status